pkg/controller/backupentry/genericactuator/types.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genericactuator
+
+import (
+	"context"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+const (
+	// EtcdBackupSecretName is the name of secret having credentials for etcd backups.
+	EtcdBackupSecretName string = "etcd-backup"
+
+	backupBucketName string = "bucketName"
+)
+
+// ProviderActuator preforms provider specific operation with BackupBucket resources.
+type ProviderActuator interface {
```

Comment:
In the generic worker actuator we call this `WorkerDelegate`, do you want to rename it to `BackupEntryDelegate` ?

Reply:
Sound good to me